File: exercises/go/practice/reverse-string/reverse_string_test.go
package reverse_string

import "testing"

func TestReverse(t *testing.T) {
	for _, test := range testCases {
		if actual := Reverse(test.input); actual != test.expected {
			t.Errorf("Reverse(%q) = %q, want %q", test.input, actual, test.expected)
		}
	}
}

var testCases = []struct {
	description string
	input       string
	expected    string
}{
	{
		description: "an empty string",
		input:       "",
		expected:    "",
	},
	{
		description: "a word",
		input:       "robot",
		expected:    "tobor",
	},
	{
		description: "a capitalized word",
		input:       "Ramen",
		expected:    "nemaR",
	},
	{
		description: "a sentence with punctuation",
		input:       "I'm hungry!",
		expected:    "!yrgnuh m'I",
	},
	{
		description: "a palindrome",
		input:       "racecar",
		expected:    "racecar",
	},
	{
		description: "an even-sized word",
		input:       "drawer",
		expected:    "reward",
	},
}

File: exercises/typescript/practice/reverse-string/tests/reverse-string.test.ts
import { expect, test } from 'vitest'
import { reverse } from '../src/reverse-string'

test('empty string', () => {
  expect(reverse('')).toBe('')
})

test('word', () => {
  expect(reverse('robot')).toBe('tobor')
})

test('capitalized word', () => {
  expect(reverse('Ramen')).toBe('nemaR')
})

test('sentence with punctuation', () => {
  expect(reverse('I\'m hungry!')).toBe('!yrgnuh m\'I')
})

test('palindrome', () => {
  expect(reverse('racecar')).toBe('racecar')
})